targets/targets.json

```diff
@@ -1292,248 +1308,160 @@
             "RESET_REASON"
         ]
     },
-    "MIMXRT1050_EVK": {
-        "supported_form_factors": [
-            "ARDUINO"
-        ],
-        "core": "Cortex-M7FD",
-        "supported_toolchains": [
-            "ARM",
-            "GCC_ARM",
-            "IAR"
+    "MCU_STM32_BAREMETAL": {
+        "inherits": [
+            "MCU_STM32"
         ],
-        "extra_labels": [
-            "NXP",
-            "MCUXpresso_MCUS",
-            "EVK",
-            "MIMXRT1050",
-            "IMX",
-            "NXP_EMAC"
+        "public": false,
+        "supported_c_libs": {
+            "arm": [
+                "small"
+            ],
+            "gcc_arm": [
+                "small"
+            ],
+            "iar": [
+                "std"
+            ]
+        },
+        "c_lib": "small",
+        "supported_application_profiles": [
```

Comment:
We should add  `"supported_application_profiles" : ["bare-metal", "full"]` to all targets in a subsequent PR.